src/main/java/com/faforever/client/coop/CoopController.java

```diff
@@ -194,6 +204,22 @@ public void initialize() {
     });
   }
 
+  private void onSearchFieldChange(String newValue){
+    leaderboardFilteredList.setPredicate(coopResultBean -> {
+      boolean foundmatch = false;
+      for(Map.Entry<String, List<String>> entry : coopResultBean.getReplay().getTeams().entrySet()){
+        foundmatch =  entry.getValue().stream().anyMatch((s) -> s.toLowerCase().contains(newValue.toLowerCase()));
+        if(foundmatch)
+          break;
+      }
+      if(newValue.isEmpty()){
+        return true;
+      }
+      return foundmatch;
```

Comment:
This is still more verbose than necessary. You call entryset but we are only using the values.
Also if newValue.isempty() we don't care about looping through the teams so we should check that first.

I would use the .values() method and the flatMap method of a stream and you should be able to make this much more readable.